src/rpc/common/Validators.cpp

```diff
@@ -212,4 +213,23 @@ CustomValidator SubscribeAccountsValidator =
         return MaybeError{};
     }};
 
+CustomValidator AMMAssetValidator =
+    CustomValidator{[](boost::json::value const& value, std::string_view key) -> MaybeError {
+        ASSERT(value.is_object(), "{}: Value must be a JSON object", key);
```

Comment:
Now I'm thinking because it is a validator it is betterer ti return `Error` here. Because input here comes from clients and clients shouldn't be able to kill clio with wrong input.

Reply:
Yea but you can't get here with non-object because we have `IfType` checks on call-site. But in general I agree that it may be that in the future someone uses it without `IfType` and then it may backfire..